### ISSUE TITLE
site_config.lua on aarch64 Linux is pointing for the wrong dep path

### ISSUE THREAD

**Issue by cosmo0920** [luarocks/luarocks#1464]
### Please read the following before submitting:
- Please do NOT submit bug reports for questions. Ask questions in the [Gitter
  chat-room](https://gitter.im/luarocks/luarocks).
- Please do NOT submit duplicate bug reports. Look through the [issue
  tracker](https://github.com/luarocks/luarocks/issues) before submitting.

### Please fill out the following:
- **Platform:**
  - Windows/Linux/MacOS.
    - Linux aarch64

- **LuaRocks version:**
  - `luarocks --version`
    - 2.4.2

- **Configuration file:**
  - Please try to produce with the default configuration.
  - If you cannot reproduce with the default configuration, please try to find
    the minimal configuration to reproduce.
  - Upload the config to a pastebin such as gist.github.com.
    - None

- **LuaRocks output from when the issue occurred:**
  - Use the `--verbose` flag.
    - None

- **Description:**
  - The steps you took in plain English to reproduce the problem.

luarocks on Linux aarch64 points into the wrong dep path. When using this command on aarch64 Linux, it fails to install dependencies.

site_config.lua

```lua
cal popen = io.popen
local site_config = {}
site_config.LUAROCKS_PREFIX=[[/usr]]
site_config.LUA_INCDIR=[[/usr/include/lua5.2]]
site_config.LUA_LIBDIR=[[/usr/local/lib]]
site_config.LUA_BINDIR=[[/usr/bin]]
site_config.LUA_INTERPRETER=[[lua5.2]]
site_config.LUAROCKS_SYSCONFDIR=[[/etc/luarocks]]
site_config.LUAROCKS_ROCKS_TREE=[[/usr/local/]]
site_config.LUAROCKS_ROCKS_SUBDIR=[[/lib/luarocks/rocks]]
site_config.LUA_DIR_SET=true
site_config.LUAROCKS_UNAME_S=(popen("uname -s"):read("*a"):gsub("\n",""))
site_config.LUAROCKS_UNAME_M=(popen("uname -m"):read("*a"):gsub("\n",""))
site_config.LUAROCKS_DOWNLOADER=[[wget]]
site_config.LUAROCKS_MD5CHECKER=[[md5sum]]
-- LUAROCKS_EXTERNAL_DEPS_SUBDIRS points to lib/x86_64-linux-gnu but, on aarch64, it should point into lib/aarch-linux-gnu
site_config.LUAROCKS_EXTERNAL_DEPS_SUBDIRS={ bin="bin", lib={ "lib", [[lib/x86_64-linux-gnu]] }, include="include" }
site_config.LUAROCKS_RUNTIME_EXTERNAL_DEPS_SUBDIRS={ bin="bin", lib={ "lib", [[lib/x86_64-linux-gnu]] }, include="include" }
return site_config
```

```log
$ uname -i
aarch64
$ [ -d /usr/lib/aarch64-linux-gnu ] && echo "Directory exists"
Directory exists
$ [ -d /usr/lib/x86_64-linux-gnu ] && echo "Directory exists"
```

**Comment by hishamhm** [comment]
@cosmo0920 LuaRocks 2.4.2 is very old; please upgrade to the latest version, and you will not have this issue.
